src/MixerBoard.cpp

```diff
@@ -527,7 +528,7 @@ void FillBufferWithSampleView(float* buffer, const ChannelSampleView& view)
    size_t copiedSamples = 0;
    for (const auto& segmentView : view)
    {
-      const auto sampleCount = segmentView.GetSampleCount();
+      const auto sampleCount = segmentView.GetSampleCount().as_size_t();
```

Comment:
Restating my lost comment:  This line is why I don't like the change.

`as_size_t` has an assertion that the value of the sample count is really in the range of a `size_t`.  I wasn't following the `@pre` comment convention yet those years ago.  It should be a `@pre` now.

And that means, there is burden of proof of satisfaction of the precondition at the point of the call.  But I don't know what the proof is here, and I don't think there is any sufficient information to prove it.

The point of commit 48aaca450b113a971a487f768c3e941accf9395a was to shift the burden of that proof instead to the places where `AudioSegmentSampleView` is constructed.

And note too, in that commit, the places where I added `@pre stretched samples in [t0, t1) can be counted in a `size_t` ` as an informal precondition, moving the burden of proof farther up.

And in the next commit after that, which is ba920717e826d4a0d6b5d81436d90f6a2e8f69df, I made those try-catches of `bad_alloc` in case the precondition was not satisfied.

Reply:
So I don't approve of the reversal of all of that caution that I put into an earlier commit.  And I must finish my examination of this pull request to see where you use the constructor and what proof of satisfaction of the precondition is possible.

Reply:
I don't see how it was safer before. I'm looking at this:
```cpp
   // Not sure how to prove satisfaction of the invariant of GetSampleView
   mSampleView = pTrack->GetSampleView(t0, t1, mayThrow);

   // Expect that the difference of t1 and t0 is the part of a track played
   // in about 1/20 second (ticks of TrackPanel timer), so this won't overflow,
   // unless stretch ratio is extremely low.
   const auto nFrames = GetNumSamplesInView(mSampleView[0]);
```

If `AudioSegmentSampleView::mLength` is `size_t`,  overflow could happen in the first statement.
If it takes a `sampleCount`, overflow could happen in  `GetNumSamplesInView`.
Either way the overflow is caught by your try-catch.

Also, this would only happen in extreme cases : the MixerBoard refreshes every 50ms. What stretch ratio do we need to squeeze more than 2^31 samples of audio in such a short amount of time ? Possible, yet again there's the try-catch.

With `sampleCount`, the risk of overflow doesn't matter for this check in the next commit:
```cpp
   if (GetNumSamples(mChannelGroupSampleViews) > maxDataLen) {
      auto msg = XO(
"Too much audio was selected. Only the first %.1f seconds of audio will be analyzed.")
         .Format(maxDataLen / mRate);
      AudacityMessageBox( msg );
   }
```

But again, I didn't realize you had changed it, and didn't revert it properly as I left your comments untouched.
We have a disagreement about whether this PR should go in or not, let's wait until that's settled to see if I have to clean that up.